chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts

```diff
@@ -13,25 +13,34 @@ import { IReconnectChatContext } from "../../reconnectchatpanestateful/interface
 import { IReconnectChatOptionalParams } from "../../reconnectchatpanestateful/interfaces/IReconnectChatOptionalParams";
 import { LiveChatWidgetActionType } from "../../../contexts/common/LiveChatWidgetActionType";
 import { TelemetryHelper } from "../../../common/telemetry/TelemetryHelper";
+import { handleAuthentication, removeAuthTokenProvider } from "./authHelper";
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const getChatReconnectContext = async (chatSDK: any, reconnectId?: string) => {
-    try {
-        if (reconnectId) {
-            const chatReconnectOptionalParams: IReconnectChatOptionalParams = {
-                reconnectId: reconnectId
-            };
-            return await chatSDK?.getChatReconnectContext(chatReconnectOptionalParams);
-        } else {
-            return await chatSDK?.getChatReconnectContext();
-        }
-    } catch (ex) {
-        TelemetryHelper.logSDKEvent(LogLevel.ERROR, {
-            Event: TelemetryEvent.GetChatReconnectContextSDKCallFailed,
-            ExceptionDetails: {
-                exception: ex
+const getChatReconnectContext = async (chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, isReconnectEnabled?: boolean, reconnectId?: string) => {
+    if (isReconnectEnabled) {
+        try {
+            if (reconnectId) {
+                const chatReconnectOptionalParams: IReconnectChatOptionalParams = {
+                    reconnectId: reconnectId
+                };
+                return await chatSDK?.getChatReconnectContext(chatReconnectOptionalParams);
+            } else {
+                // set auth token to chat sdk to get reconnectId for auth chat
+                await handleAuthentication(chatSDK, chatConfig, getAuthToken);
+                const reconnectChatContext = await chatSDK?.getChatReconnectContext();
+                // remove auth token after reconnectId is fetched
+                // this will be reset later at start chat
+                removeAuthTokenProvider(chatSDK);
```

Comment:
Seems we're moving the token, does it mean we'll get a new token for authentication? If yes, is reconnect working in the token changed for `startChat`?

Reply:
The token will be in a global variable in script level, so it will still be the same token. But if it is expired, we will get a new token. My understanding is that token is not relevant for reconnecting, and that it reconnects to previous chat based on reconnectId